stock_release_channel/models/stock_release_channel.py

```diff
@@ -0,0 +1,633 @@
+
+        all_picking_ids = [
+            pid for picking_ids in picking_ids_per_field.values() for pid in picking_ids
+        ]
+        data = self.env["stock.move"].read_group(
+            # TODO for now we do estimates, later we may improve the domains per
+            # field, but now we can run one sql query on stock.move for all fields
+            [("picking_id", "in", all_picking_ids), ("state", "!=", "cancel")],
+            ["picking_id"],
+            ["picking_id"],
+        )
+        move_count = {
+            row["picking_id"][0]: row["picking_id_count"]
+            for row in data
+            if row["picking_id"]
+        }
+        for field, __ in domains.items():
+            move_field = field.replace("picking", "move")
+            for record in self:
+                picking_ids = picking_ids_per_field.get((record.id, field), [])
+                move_estimate = sum(
+                    move_count.get(picking_id, 0) for picking_id in picking_ids
+                )
+                record[move_field] = move_estimate
+
+        for record in self:
+            record.count_picking_full_progress = (
+                record.count_picking_release_ready
+                + record.count_picking_released
+                + record.count_picking_done
+            )
+
+    def _query_get_chain(self, pickings):
+        """Get all stock.picking before an outgoing one
+
+        Follow recursively the move_orig_ids.
+        Outgoing picking ids are excluded
+        """
+        query = """
+        WITH RECURSIVE
+        pickings AS (
+            SELECT move.picking_id,
+                   true as outgoing,
+                   ''::varchar as state,  -- no need it, we exclude outgoing
+                   move.id as move_orig_id
+            FROM stock_move move
+            WHERE move.picking_id in %s
+
+            UNION
+
+            SELECT move.picking_id,
+                   false as outgoing,
+                   picking.state,
+                   rel.move_orig_id
+            FROM stock_move_move_rel rel
+            INNER JOIN pickings
+            ON pickings.move_orig_id = rel.move_dest_id
+            INNER JOIN stock_move move
+            ON move.id = rel.move_orig_id
+            INNER JOIN stock_picking picking
+            ON picking.id = move.picking_id
+        )
+        SELECT DISTINCT picking_id, state FROM pickings
+        WHERE outgoing is false;
+        """
+        return (query, (tuple(pickings.ids),))
+
+    def _compute_picking_chain(self):
+        self.env["stock.move"].flush_model(
+            ["move_dest_ids", "move_orig_ids", "picking_id"]
+        )
+        self.env["stock.picking"].flush_model(["state"])
+        for channel in self:
+            domain = self._field_picking_domains()["count_picking_released"]
+            domain += [("release_channel_id", "=", channel.id)]
+            released = self.env["stock.picking"].search(domain)
```

Comment:
To avoid several search DB queries, I would have put this outside the loop and then filter on the release channel